controller/src/controller/models/domain/imagepolicy.py

```diff
@@ -0,0 +1,150 @@
+"""Models for image display policy."""
+
+import datetime
+from typing import Annotated, Any
+
+from pydantic import BaseModel, BeforeValidator, ConfigDict, Field
+from semver.version import VersionInfo
+
+__all__ = ["IndividualImageClassPolicy", "RSPImagePolicy"]
+
+
+def _empty_str_is_none(inp: Any) -> Any:
+    if isinstance(inp, str) and inp == "":
+        return None
+    return inp
+
+
+class IndividualImageClassPolicy(BaseModel):
+    """Policy for images to display within a given class.
+
+    The policy has both a 'number' and an 'age' field.
+
+    'number' means: display that many of whatever image class this is
+    attached to.  `-1` or `None` are interpreted as "do not filter
+    (i.e. show all of this image class)" and `0` means "display no
+    images of this class."  This has historically been the only filter
+    option.
+
+    'age' means: display any items of the class whose age is the
+    specified age or less.  This age is a duration, specified by a
+    string as accepted by Safir's HumanTimeDelta.  The empty string
+    means "do not filter this class at all."
+
+    'cutoff_version' means: display any items of this class whose semantic
+    version is this large or larger.  This only makes sense for dailies,
+    weeklies, releases, release candidates, and experimentals derived from
+    one of those builds.
+
+    'filter_empty' governs the behavior when a policy is applied to an image
+    that does not have data for that category (e.g. an 'age' policy exists,
+    but the image age is 'None' because it could not be determined).  If
+    'filter_empty' is 'True', the image will not be displayed, and if 'False'
+    the image will be displayed.
```

Comment:
I would just pick a behavior and delete the option until we have a use case for being able to configure this. `False` seems like a reasonable approach; let's see if it causes any problems.